### PR TITLE
STITCH-1733 Create JS Admin SDK

### PR DESCRIPTION
**Pull request by jsflax**

### CODE REVIEW

**Comment by edaniels** [comment]
@mpobrien @UnicodeSnowman heads up

**Comment by edaniels** [comment]
also @makesitgo since I know he appreciates TS

File: packages/core/admin-client/src/resources/ValuesResource.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+// / Resource for a list of users of an application
+export class ValuesResource extends BasicResource<ValuesRoutes>
+  implements
+    Listable<ValueResponse, ValuesRoutes>,
+    Creatable<ValueCreator, ValueResponse, ValuesRoutes> {
+  public readonly codec = new ValueResponseCodec();
+  public readonly creatorCodec = new ValueCreatorCodec();
+
+  public create: (data: ValueCreator) => Promise<ValueResponse>;
+  public list: () => Promise<ValueResponse[]>;
```
[the remark this answers is not in the document]
Comment (adamchel) [reply packages/core/admin-client/src/resources/ValuesResource.ts]: Eric's suggestion would probably be the most semantically correct way to do it, but you could also just make the fields optional if they don't appear in partial responses.

File: packages/core/testutils/src/BaseStitchIntTestHarness.ts [hunk 2]
```diff
@@ -77,24 +79,24 @@ export default abstract class BaseStitchIntTestHarness {
       this.apps.map(app => {
         app.remove();
       })
-    ).then(() => this.adminClient.logout());
+    ).then(() => this.adminClient.adminAuth.logout());
```
Comment (adamchel) [inline packages/core/testutils/src/BaseStitchIntTestHarness.ts]: [nit] this is an example of why I think `adminAuth` should be renamed to just `auth`.

File: packages/core/admin-client/__tests__/StitchAdminClientIntTests.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
+import { ObjectId } from "bson";
+import { UserPasswordCredential } from "mongodb-stitch-core-sdk";
+import { ProviderConfigCodec, Userpass } from "../src/configs/AuthProviderConfigs";
+import { AppResource, AppResponse, AppsResource, StitchAdminClient, AwsS3, AwsS3RuleCreator, AwsS3Actions } from "../src/index";
+import { StitchAdminClientConfiguration } from "../src/StitchAdminClientConfiguration";
+
+async function createApp(): Promise<AppResponse> {
+    return appsResource.create({name: "fun-app"})
```
Comment (adamchel) [inline packages/core/admin-client/__tests__/StitchAdminClientIntTests.ts]: [nit] should probably be a generated objectId instead of `fun-app`

**Pull request closed without merging**